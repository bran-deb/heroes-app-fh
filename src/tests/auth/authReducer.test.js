import { authReducer } from "../../auth/authReducer"
import { types } from "../../types/types"
import { shallow } from 'enzyme'

describe('Pruebas en <authReducer/>', () => {
    // const state = {
    //     name: 'jairo',
    //     logged: true
    // }
    test('debe de mostrarse correctamente', () => {
        const wrapper = shallow(<authReducer />)
        expect(wrapper).toMatchSnapshot()
    })

    test('debe de retornar el estado por defecto', () => {
        const state = authReducer({ logged: false }, {})
        expect(state).toEqual({ logged: false })
    })

    test('debe de autenticar y colocar el name del usuario', () => {
        const action = {
            type: types.login,
            payload: {
                name: 'hernando'
            }
        }
        const state = authReducer({ logged: false }, action)
        expect(state).toEqual({
            logged: true,
            name: 'hernando'
        })
    })

    test('debe de borrar el name del usuario y logged false', () => {
        const action = {
            type: types.logout,
            payload: {
                name: ''
            }
        }
        const state = authReducer({ logged: true, name: 'pedrales' }, action)
        expect(state).toEqual({
            logged: false
        })
    })

})
