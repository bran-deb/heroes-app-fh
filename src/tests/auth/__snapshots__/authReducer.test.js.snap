// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <authReducer/> debe de mostrarse correctamente 1`] = `<authReducer />`;
